### PR TITLE
[SDK] Simplify env config usage

### PR DESCRIPTION
**Pull request by yaronha**
* use `--env-file <env file path>` in mlrun run/build/deploy/project CLI commands to load config and credential env vars from file
* set the `MLRUN_SET_ENV_FILE=<env file path>`  env var to point to a default env file (will be loaded on import)
* when the `MLRUN_DBPATH` points to remote iguazio cluster and the `V3IO_API` and/or `V3IO_FRAMESD` vars are not set they will be inferred from the DBPATH
* add default env file template in the Jupyter container `~/env`  (allow quick setup of remote demos)

### CODE REVIEW

File: dockerfiles/jupyter/Dockerfile [hunk 1]
```diff
@@ -41,7 +42,8 @@ ENV MLRUN_ARTIFACT_PATH=$HOME/data \
     MLRUN_HTTPDB__DATA_VOLUME=$HOME/data \
     MLRUN_HTTPDB__DSN='sqlite:////home/jovyan/data/mlrun.db?check_same_thread=false' \
     MLRUN_HTTPDB__LOGS_PATH=$HOME/data/logs \
-    MLRUN_PVC_MOUNT=nfsvol:/home/jovyan/data
+    MLRUN_PVC_MOUNT=nfsvol:/home/jovyan/data \
+    MLRUN_SET_ENV_FILE=$HOME/env
```
Comment (Hedingber) [inline dockerfiles/jupyter/Dockerfile:46]: I suggest to rename to `MLRUN_CONFIG_FILE` or `MLRUN_ENV_FILE` if you feel like separating between the config and the env

File: dockerfiles/jupyter/env [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# default env vars, will be loaded once MLRun imports/starts
```
Comment (Hedingber) [inline dockerfiles/jupyter/env:1]: The standard seems to be to name the file `.env`

File: mlrun/config.py [hunk 3]
```diff
@@ -650,17 +654,27 @@ def read_env(env=None, prefix=env_prefix):
             cfg = cfg.setdefault(name, {})
         cfg[path[0]] = value
 
+    env_db_path = env.get("MLRUN_DBPATH", "")
+    is_remote_mlrun = env_db_path.startswith("https://mlrun-api.")
```
Comment (Hedingber) [inline mlrun/config.py:658]: This heuristic is not good enough, we may have the db path with this prefix in a marketplace instance of the kit (which doesn't have v3io)

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: dockerfiles/jupyter/env (hunk 2 above)
Comment (yaronha) [reply dockerfiles/jupyter/env:1]: i think it wont be visible in Jupy w `.` , saw Azure demo it was `env`

**Review by Hedingber — APPROVED** [review]

**Pull request merged**